mixer/pkg/api/grpcServer.go

```diff
@@ -216,16 +221,24 @@ var reportResp = &mixerpb.ReportResponse{}
 
 // Report is the entry point for the external Report method
 func (s *grpcServer) Report(ctx context.Context, req *mixerpb.ReportRequest) (*mixerpb.ReportResponse, error) {
+	qrs := atomic.LoadInt64(&s.queuedRequests)
+	if qrs > s.queueLength {
```

Comment:
I think that's too strict, it doesn't allow for burstiness. How about something like queueLength * 8?

Reply:
Agreed. This was more PoC than anything else. Wanted to have *some* rubric that was easy to test initially for discussion.

Reply:
Reports arrive in large batches. At this point in the code, we want to make a judgement about whether we can service 1000 new reports in a reasonable amount of time.  Treating all requests equally is not correct. We could assume that all requests cost the same and do it here, but the rejection should be based on response time violations.

Reply:
@mandarjog are you suggesting that we shed current load based on prior response times?